### DISCUSSION TITLE
Gatsby-Plugin-Image Default Inline Styles

### DISCUSSION THREAD

**Discussion by Harry-Dang** [gatsbyjs/gatsby#36600] [Help]
I'm teaching myself full-stack development and I decided to create a portfolio page for myself using Gatsby.

In a tutorial I was following, they said to use the Gatsby-Plugin-Image plugin for images for better web optimization. I wanted to create a responsive image gallery using flexboxes ([as seen here](https://css-tricks.com/adaptive-photo-layout-with-flexbox/)), however, I noticed in the DOM that using `GatsbyImage` gives default inline styles for `width`, `height`, and `object-fit`.

Is there a way to remove these? Or what would be the best way to override these styles? I can easily override the `object-fit` and `height`, but how should I override the width when I want to keep the aspect ratio?

Thanks

**Comment by LekoArts** [comment]
Hi!

The styles are there to not cause layout shifts when the site is loaded, improving the Lighthouse score. These are the safe defaults, however you can override them with `imgStyle` (https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image#shared-props)

From the `gatsbyImageData` query you get back the width and height so you can calculate the aspect ratio on your own.

I've built different kinds of image grids with gatsby-plugin-image in case you want to see the CSS:
- https://jodie.lekoarts.de/
- https://emma.lekoarts.de/
- https://emilia.lekoarts.de/

**Reply by Harry-Dang** [reply]
Thanks

> From the `gatsbyImageData` query you get back the width and height so you can calculate the aspect ratio on your own.

I tried this and it works okay, however, because we're setting the width, it no longer uses the flexbox's flex-grow width to fill the row which is the look I am trying to go for as seen in the tutorial I had linked. How should I go about calculating this width?

> I've built different kinds of image grids with gatsby-plugin-image in case you want to see the CSS:

I noticed in your examples, the `<div data-gatsby-image-wrapper ... >` has no inline style. How did you get it to do that? Is it because it is in a grid container?
